site/src/request_handlers/github.rs

```diff
@@ -153,47 +169,82 @@ async fn handle_rust_timer(
         main_client,
         ci_client,
         issue.number,
-        build_captures(&comment.body).map(|(commit, _)| commit),
+        valid_build_cmds.iter().map(|c| c.sha),
     )
     .await?;
 
     Ok(github::Response)
 }
 
-/// Parses the first occurrence of a `@rust-timer queue <...>` command
+/// Parses the first occurrence of a `@rust-timer queue <shared-args>` command
 /// in the input string.
 fn parse_queue_command(body: &str) -> Option<Result<QueueCommand, String>> {
-    let prefix = "@rust-timer";
-    let bot_line = body.lines().find_map(|line| {
-        line.find(prefix)
-            .map(|index| line[index + prefix.len()..].trim())
-    })?;
-
-    let args = bot_line.strip_prefix("queue").map(|l| l.trim())?;
-    let mut args = match parse_command_arguments(args) {
+    let args = get_command_lines(body, "queue").next()?;
+    let args = match parse_command_arguments(args) {
         Ok(args) => args,
         Err(error) => return Some(Err(error)),
     };
-    let mut cmd = QueueCommand {
+    let params = match parse_benchmark_parameters(args) {
+        Ok(params) => params,
+        Err(error) => return Some(Err(error)),
+    };
+
+    Some(Ok(QueueCommand { params }))
+}
+
+/// Parses all occurrences of a `@rust-timer build <shared-args>` command in the input string.
+fn parse_build_commands(body: &str) -> impl Iterator<Item = Result<BuildCommand, String>> {
+    get_command_lines(body, "build").map(|line| {
+        let mut iter = line.splitn(2, ' ');
+        let Some(sha) = iter.next().filter(|s| !s.is_empty() && !s.contains('=')) else {
+            return Err("Missing SHA in build command".to_string());
+        };
+
+        let sha = sha.trim_start_matches("https://github.com/rust-lang/rust/commit/");
+        let args = iter.next().unwrap_or("");
+        let args = parse_command_arguments(args)?;
+        let params = parse_benchmark_parameters(args)?;
+        Ok(BuildCommand { sha, params })
+    })
+}
+
+fn get_command_lines<'a: 'b, 'b>(
```

Comment:
I don't think you need 'b here?

Reply:
It's not like I like to add random lifetime parameters everywhere xD It wasn't compiling without it. But apparently it was just me being tired on the bus. This seems to work:
```rust
fn get_command_lines<'a>(body: &'a str, command: &'a str) -> impl Iterator<Item = &'a str> {
    let prefix = "@rust-timer";
    body.lines()
        .filter_map(move |line| {
            line.find(prefix)
                .map(|index| line[index + prefix.len()..].trim())
        })
        .filter_map(move |line| line.strip_prefix(command))
        .map(|l| l.trim_start())
}
```
Thanks!